### ISSUE TITLE
Move _$_ from Reflection.Base to Foundations.Function

### ISSUE THREAD

**Issue by ecavallo** [agda/cubical#894]
It doesn't have anything to do with reflection (don't know why I put it there in the first place) and is generally useful.

The obstacle is that different functions called _$_ are defined elsewhere in the library, namely in Algebra.Ring.Base and Data.Graph.Base. We'd have to rename these.

**Comment by mzeuner** [comment]
Maybe we could use something like `_$r_` for ring morphisms?

**Comment by ecavallo** [comment]
Yeah, I think there's already a ``_$a_`` for algebras or something like that. For the graph homomorphisms it might be best to change it to match the Algebra conventions for homs.

**Comment by felixwellen** [comment]
I think I defined ```_$a_``` but ended up not using it. But now that I think of it, I'm actually for using it and similar functions.
